nixos/modules/system/boot/stage-1.nix

```diff
@@ -251,7 +251,8 @@ let
     postInstall = ''
       echo checking syntax
       # check both with bash
-      ${pkgs.bash}/bin/sh -n $target
+      ${pkgs.buildPackages.bash}/bin/sh -n $target
+    '' + optionalString (pkgs.stdenv.hostPlatform == pkgs.stdenv.buildPlatform) ''
       # and with ash shell, just in case
       ${extraUtils}/bin/ash -n $target
```

Comment:
I think the ash check is actually more important than the bash check. Maybe we can do one check with ```${pkgs.buildPackages.busybox}/bin/ash -n $target```.